### DISCUSSION TITLE
Add `nestedRoutes` to `@react-router/fs-routes`

### DISCUSSION THREAD

**Discussion by okalil** [remix-run/react-router#12647] [Proposals]
The @react-router/fs-routes package should export a foldersRoutes function, that create route config from the file system using a convention that matches Remix v1's route file naming, as an alternative to flatRoutes.

**Comment by brophdawg11** [comment]
Linking the PR with a POC implementation we can re-open if this moves forward: https://github.com/remix-run/react-router/pull/12648

**Comment by brophdawg11** [comment]
To quote the new [Governance doc](https://github.com/remix-run/react-router/blob/main/GOVERNANCE.md#design-goals):

> **Routing and Data Focused.** Focus on core router-integrated/router-centric APIs and avoid adding first-class APIs that can be implemented in user-land

I think my gut reaction here is that this might be better off as a community package.  We moved away from this convention in v2 and carved off [`@remix-run/v1-route-convention`](https://www.npmjs.com/package/@remix-run/v1-route-convention) for folks who didn't want to migrate to flat routes.  Can that package be used with `@react-router/remix-routes-option-adapter`?

We specifically put that in a separate repo because it wasn't the primary approach anymore and because it was mostly considered stable/frozen and wasn't really going to get the same level of maintenance as the other packages.  I think the same holds true today - we don't necessarily want to bring this code in-house again because then it's one more thing we need to document/maintain.

We moved to `routes.ts` to _avoid_ trying to solve file-based-routing and so folks could much more easily write their own conventions.  `flatRoutes` exists again as a _very_ simple first-class convention but also mostly for migration from v2.  I don't think we _want_ to maintain other first-class conventions since we've found that there's simply just not a convention that makes enough poeple happy to so we want out of the convention game entirely :).

**Reply by okalil** [reply]
Agreed. Combining `@remix-run/v1-route-convention` with `@react-router/remix-config-routes-adapter` was suggested a while ago [in this Discord thread](https://discordapp.com/channels/770287896669978684/770287896669978687/1303019712304644250). We could mention this solution in the "Upgrading from Remix" doc to help users migrating from Remix v1 to React Router v7.

**Reply by brophdawg11** [reply]
yeah that's a good idea - do you want to open a PR to update that doc?

**Reply by okalil** [reply]
Linking here: https://github.com/remix-run/react-router/pull/13921
